repo-client/src/main/kotlin/io/lionweb/repoclient/LionWebClient.kt

```diff
@@ -0,0 +1,734 @@
+                .url(url)
+                .considerAuthenticationToken()
+                .post(EMPTY_REQUEST)
+                .build()
+        OkHttpClient().newCall(request).execute().use { response ->
+            if (response.code != HttpURLConnection.HTTP_OK) {
+                throw RuntimeException("DB initialization failed, HTTP ${response.code}: ${response.body?.string()}")
+            }
+        }
+    }
+
+    // Partitions
+
+    fun createPartition(node: Node) {
+        if (node.children.isNotEmpty()) {
+            throw IllegalArgumentException("When creating a partition, please specify a single node")
+        }
+        treeStoringOperation(node, "createPartitions")
+    }
+
+    fun deletePartition(node: Node) {
+        deletePartition(node.id ?: throw IllegalStateException("Node ID not specified"))
+    }
+
+    fun deletePartition(nodeID: String) {
+        val body: RequestBody = "[\"$nodeID\"]".toRequestBody(JSON)
+        val request: Request =
+            Request.Builder()
+                .url("http://$hostname:$port/bulk/deletePartitions")
+                .considerAuthenticationToken()
+                .post(body)
+                .build()
+        httpClient.newCall(request).execute().use { response ->
+            if (response.code != HttpURLConnection.HTTP_OK) {
+                val body = response.body?.string()
+                if (debug) {
+                    println("  Response: ${response.code}")
+                    println("  Response: $body")
+                }
+                throw RuntimeException("Request failed with code ${response.code}: $body")
+            }
+        }
+    }
+
+    private fun Request.Builder.considerAuthenticationToken(): Request.Builder {
+        return if (authorizationToken == null) {
+            this
+        } else {
+            this.addHeader("Authorization", authorizationToken)
+        }
+    }
+
+    fun getPartitionIDs(): List<String> {
+        val url = "http://$hostname:$port/bulk/listPartitions"
+        val request: Request =
+            Request.Builder()
+                .url(url)
+                .considerAuthenticationToken()
+                .addHeader("Accept-Encoding", "gzip")
+                .post(EMPTY_REQUEST)
+                .build()
+        httpClient.newCall(request).execute().use { response ->
+            if (response.code == HttpURLConnection.HTTP_OK) {
+                val data =
+                    (response.body ?: throw IllegalStateException("Response without body when querying $url")).string()
+                return processChunkResponse(data) {
+                    val chunk = LowLevelJsonSerialization().deserializeSerializationBlock(it)
+                    chunk.classifierInstances.mapNotNull { it.id }
+                }
+            } else {
+                throw RuntimeException("Got back ${response.code}: ${response.body?.string()}")
+            }
+        }
+    }
+
+    // Nodes
+
+    fun retrieve(
```

Comment:
Looks like this function is identical to the next retrieve function. 
Can be replaced by next function with `[rootId]` as parameter.
function body then simply is:
`return this.retrieve([rootId, withProxyParent, retrievalMode)[0]`

Reply:
Oh, wow. A lot of duplicated code. I fixed it. Thank you!